### ISSUE TITLE
Confusing "Variable identifier expected" error when compiling

### ISSUE THREAD

**Issue by ProxyPlayerHD** [cc65/cc65#1666]
a bit of background: i came across issue #171 a few weeks ago and thought "why not try to port libfixmath to CC65"? so that's what i've been doing.
i wrote the Multiplication and Division routines in assembly as CC65 doesn't support 64-bit data types, but when it comes to the Trigonometric Functions i just took the original C code and adjusted it a bit (ie remove most of not all #ifdef statements).

now the actual problem is: sin is not compiling and i cannot understand why. the code works on any online C compiler i tried... here:

```
// ---------------------------------------------------------------------------
// fxtrig.c
// ---------------------------------------------------------------------------
//
// Fixed Point Trigonometric Functions
//

#include <stdint.h>
#include "FixedMath.h"

fix16_t fxsin(fix16_t inAngle){
	fix16_t tempAngle = inAngle % fx_pi2;
	
	if(tempAngle > fx_pi){
		tempAngle = tempAngle - fx_pi2;
	}
	else if(tempAngle < -fx_pi){
		tempAngle = tempAngle + fx_pi2;
	}
	
	fix16_t tempAngleSq = fxmul(tempAngle, tempAngle);		// This is line 21
	
	// Fast implementation, accuracy of ~2.3%
	fix16_t tempOut = fxmul(-13, tempAngleSq) + 546;		// This is line 24
	tempOut = fxmul(tempOut, tempAngleSq) - 10923;
	tempOut = fxmul(tempOut, tempAngleSq) + 65536;
	return fxmul(tempOut, tempAngle);
}
```
fx_pihalf, fx_pi, and fx_pi2, and the fix16_t data type are defined in an included header file:
```
typedef int32_t	fix16_t;

static const fix16_t fx_pihalf	= 0x0001921F;
static const fix16_t fx_pi	= 0x0003243F;
static const fix16_t fx_pi2	= 0x0006487E;
```
replacing "static const"s with "#define"s doesn't change anything.
the command: `CC65 -t none -O --cpu 65C02 -I Include -o Temp\fxtrig.a65 fxtrig.c`
and the errors:
```
fxtrig.c:21: Error: Variable identifier expected
fxtrig.c:21: Error: ';' expected
fxtrig.c:21: Error: Undefined symbol: 'tempAngleSq'
fxtrig.c:24: Error: Variable identifier expected
fxtrig.c:24: Error: ';' expected
fxtrig.c:24: Error: Undefined symbol: 'tempOut'
6 errors and 0 warnings generated.
```

commenting out the whole `if(){...} else if(){...}` block somehow makes it work...
why? what exactly am i missing here?

**Comment by Compyx** [comment]
In C89 you cannot declare variables anywhere in a block, only at the start of a block. So I'd move the declarations of `TempAngleSq` and `tempOut` to the top of the block (above `fix16_t tempAngle = inAngle % fx_pi2;`, because that assignment contains code to initialize the tempAngle variable, meaning it must be the last variable declaration in the block).

**Comment by ProxyPlayerHD** [comment]
oooh, that explains why removing the if-else blocks fixed it, as then all the variable declarations where together at the start of the function.
still a weird limitation, but i guess that's sadly kinda expected from C89...

anyways, Thanks! It appears to be working, just a few more functions and i'll make a repo for the library so anyone can use it!

<img width="614" alt="ttermpro_2022-02-13_19-35-10" src="https://user-images.githubusercontent.com/13576843/153769457-ac5fbb18-532e-44a3-bcbd-c24380690e03.png">
(Tested on a 20MHz 65c02)

**Comment by mrdudz** [comment]
what kind of license is this lib using? native fixed point support is something i can see added to the cc65 repo (but obviously only if the license permits)

**Comment by ProxyPlayerHD** [comment]
well, i wrote most of it from scratch (besides the trig. stuff), and haven't made it public yet, so i'd had to choose a license once i make the repo.
i'll probably choose the `MIT License`, since that's also what libfixmath uses.

i will make a post on here when i upload everything.

**Comment by polluks** [comment]
> still a weird limitation, but i guess that's sadly kinda expected from C89...

@ProxyPlayerHD You don't have to change the code, simply use `--standard c99`.

**Comment by mrdudz** [comment]
Yes. It can't be said often enough. cc65 is not a c99 compiler, it's a c89 compiler with some c99 things

**Comment by ProxyPlayerHD** [comment]
even if that switch did exist i don't think i'd want to use it as it just means anyone who wants to run the same code would also need to enable that switch.

either way, i finished and uploaded the whole library and source code to this repo: https://github.com/ProxyPlayerHD/cc65-FixedMath (i also choose the GPL-3.0 License instead of MIT, but it should still allow people to implement it into cc65 if they want)
it should be fully functional, but it has no built-in print function or string conversion, so you'd have to make your own for whatever target you choose.
maybe i'll upload my (somewhat) hardware independent Serial library some time later, since it does have a fixed point print function.

also, fxtoint, fxfromint, fxmul, and fxdiv are written in 65c02 assembly, so the code needs to be ported to the NMOS 6502 with some conditional directives. their code overall needs a cleanup since i just wrote it to be functional, not fast or pretty.
so i still got some work to do, but now other people on the internet can try and help as well
